graylog2-server/src/main/java/org/graylog2/rest/resources/cluster/ClusterSystemJobResource.java

```diff
@@ -91,4 +90,31 @@ public SystemJobSummary getJob(@ApiParam(name = "jobId", required = true) @PathP
 
         throw new NotFoundException("System job with id " + jobId + " not found!");
     }
+
+    @DELETE
+    @Path("{jobId}")
+    @Timed
+    @ApiOperation(value = "Cancel job with the given ID")
+    @Produces(MediaType.APPLICATION_JSON)
+    public SystemJobSummary cancelJob(@ApiParam(name = "jobId", required = true) @PathParam("jobId") @NotEmpty String jobId) throws IOException {
+        final Optional<Response<SystemJobSummary>> summaryResponse = nodeService.allActive().entrySet().stream()
+                .map(entry -> {
+                    final RemoteSystemJobResource resource = remoteInterfaceProvider.get(entry.getValue(),
+                            this.authenticationToken, RemoteSystemJobResource.class);
+                    try {
+                        return resource.delete(jobId).execute();
+                    } catch (IOException e) {
+                        LOG.warn("Unable to fetch system jobs from node {}:", entry.getKey(), e);
+                        return null;
+                    }
+                })
+                .filter(response -> response != null && response.isSuccessful())
+                .findFirst(); // There should be only one job with the given ID in the cluster. Just take the first one.
+
+        if (summaryResponse.isPresent()) {
```

Comment:
Could be shorter with [`Optional#orElseThrow()`](https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#orElseThrow-java.util.function.Supplier-):
```java
return summaryResponse.orElseThrow(() -> new NotFoundException("System job with ID <" + jobId + "> not found!")).body();
```